src/anonymous-credentials/schema.ts

```diff
@@ -299,24 +301,22 @@ export class CredentialSchema extends Versioned {
     return encoded;
   }
 
-  static validate(schema: object) {
-    // Following 2 fields could have been implicit but being explicit for clarity
-    this.validateStringType(schema, CRED_VERSION_STR);
-    this.validateStringType(schema, SCHEMA_STR);
-
-    if (schema[STATUS_STR] !== undefined) {
-      this.validateStringType(schema[STATUS_STR], REGISTRY_ID_STR);
-      this.validateStringType(schema[STATUS_STR], REV_CHECK_STR);
-      this.validateStringType(schema[STATUS_STR], REV_ID_STR);
+  static validate(schema: any) {
+    const schemaStatus = schema.properties[STATUS_STR];
```

Comment:
yeah so feel free to argue against this change, but my thinking is this:
why do we require the fields to be defined in the schema itself if it should always be a string type with that name? it just makes using already defined schemas a bit more of a pain because one then has to modify it to have these values. i think its better that they are implicit

was also thinking the same about the schema status object

Reply:
Because then either credential/presentation have to know that there are 2 extra fields to account for or schema class has to know consider adding those fields to the output of `flatten` and maybe account for in other places as well.I prefer to avoid such leaky abstraction.

> t just makes using already defined schemas a bit more of a pain because one then has to modify it to have these values

You don't have to modify them. Just extend output of `CredentialSchema.essential()` with your schema.

> was also thinking the same about the schema status object

Similar answer as above, schema would define things exactly the way they are without making other objects aware of the what defaults/implicits are.

Reply:
>You don't have to modify them. Just extend output of CredentialSchema.essential() with your schema.

i can achieve this in the SDK for when we parse/load the schema object, but otherwise itd have to be modified by extension. ill just revert this change for sake of brevity

Reply:
however i dislike credentialSchema being a string, would much rather it be an object with two properies: `id` and `type` - ID can be a data url that represents json. for example:
data:application/json;base64,ew0KICAgICJtYWx0X3R5cGUiOiAibG9nIiwNCiAgICAibWFsdF9kYXRhIjogIldvdywgdSByIGFsbW9zdCB0aGVyZSA6TyINCn0=

and type can be something like `DockJSONSchema2022`

better for jsonld this way, and conforms to the w3c jsonld spec

Reply:
the change was reverted, we can discuss the above idea of object for schema using data uri in another PR

Reply:
> conforms to the w3c vc jsonld spec

Can you please point me to it?

> allowing you to embed the schema

Embed where?

bas64 is going to increase the size by a third. And the schema is always a JSON so i don't see the point of base64 encdoding.

Reply:
base64 is optional, dont worry too much on that for this PR

reference: https://www.w3.org/TR/vc-data-model/#data-schemas

```
The value of the credentialSchema [property](https://www.w3.org/TR/vc-data-model/#dfn-property) MUST be one or more data schemas that provide [verifiers](https://www.w3.org/TR/vc-data-model/#dfn-verifier) with enough information to determine if the provided data conforms to the provided schema. Each credentialSchema MUST specify its type (for example, JsonSchemaValidator2018), and an id [property](https://www.w3.org/TR/vc-data-model/#dfn-property) that MUST be a [URI](https://www.w3.org/TR/vc-data-model/#dfn-uri) identifying the schema file. The precise contents of each data schema is determined by the specific type definition.
```

that URI can be a data uri, like above. as for embed where, embed in the credential as you intend. only difference is instead of:

```
{
  ...,
  credentialSchema: "string"
}
```

it would be

```
{
  ...,
  credentialSchema: {
    "id": "data uri string",
    "type": "string"
  }
```